Server/DiscordBot.cs

```diff
@@ -88,9 +98,28 @@ private static List<string> SplitMessage(string message, int maxSizePerElem = 20
             Logger.Info(
                 $"Discord bot logged in as {DiscordClient.CurrentUser.Username}#{DiscordClient.CurrentUser.Discriminator}");
             Reconnecting = false;
-            string mentionPrefix = $"{DiscordClient.CurrentUser.Mention} ";
+            string mentionPrefix = $"{DiscordClient.CurrentUser.Mention}";
             DiscordClient.MessageCreated += async (_, args) => {
-                if (args.Author.IsCurrent) return;
+                if (args.Author.IsCurrent) return; //dont respond to commands from ourselves (prevent "sql-injection" esq attacks)
+                //prevent commands via dm and non-public channels
+                if (Config.CommandChannel == null) {
+                    if (!warnedAboutNullLogChannel) {
+                        Logger.Warn("You probably should set your CommandChannel in settings.json");
+                        warnedAboutNullLogChannel = true;
+                    }
+                    if (args.Channel is DiscordDmChannel) {
+#if SEND_RESP_TO_BAD_REQ_DM
+                        await args.Message.RespondAsync("DM's are not valid for running commands or communication to the bot. (Your command was not processed).");
+#endif
+                        return;
+                    }
+                }
+                else {
+                    ulong chId = ulong.Parse(Config.CommandChannel);
+                    if (args.Channel.Id != chId)
+                        return;
```

Comment:
Cache the command channel object on Discord client login, just like the log channel is cached.

Reply:
done